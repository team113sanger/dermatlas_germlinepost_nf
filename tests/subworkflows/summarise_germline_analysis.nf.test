nextflow_workflow {

    name "Test Workflow GERMLINE_COHORT_ANALYSIS"
    script "subworkflows/summarise_germline_analysis.nf"
    workflow "GERMLINE_COHORT_ANALYSIS"

    test("Should run without failures") {

        when {
        params {
            load("$baseDir/tests/testdata/test_params.json")
        }
            workflow {
                """
                input[0] =  Channel.of(tuple([study_id: "TBC"],
                            file("/lustre/scratch124/nf-test/germline/6676_cohort_raw_snps.marked.target.pass.vep.vcf.gz"), 
                            file("/lustre/scratch124/nf-test/germline/6676_cohort_raw_snps.marked.target.pass.vep.vcf.gz.tbi")))
                input[1] =  Channel.of(tuple([study_id: "TBC"],
                            file("/lustre/scratch124/nf-test/germline/6676_cohort_indel_raw.marked.target.pass.vep.vcf.gz"),
                            file("/lustre/scratch124/nf-test/germline/6676_cohort_indel_raw.marked.target.pass.vep.vcf.gz.tbi")))
                input[2] = "GRCh38"
                input[3] = "gnomAD_AF"
                input[4] = file("/lustre/scratch124/resources/dermatlas/germline/Cancer_national_genomic_test_directory_v7.2_June_2023_gene_smv_summary.tsv")
                input[5] = file("/lustre/scratch124/resources/dermatlas/COSMIC/cancer_gene_census.v97.genes.tsv")
                input[6] = file("/lustre/scratch124/resources/dermatlas/germline/FLAG_genes_maftools.tsv")
                input[7] = file("/lustre/scratch124/resources/dermatlas/ensembl/dermatlas_noncanonical_transcripts_ens103.tsv")
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
