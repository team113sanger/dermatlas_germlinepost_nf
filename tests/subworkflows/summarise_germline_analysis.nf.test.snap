{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "TBC_germline.keep.gnmadF.cosmic.maf.gz:md5,648f6be155cf3ab393ad8c01484d4595",
                        "TBC_germline.keep.gnmadF.cosmic.pathogenic.maf.gz:md5,5ce35dcb7fb36f529ce2d05955ec53a5",
                        "TBC_germline.keep.gnmadF.cosmic.pathogenic.pdf:md5,bf90eefe330b466d2da5fd79a415138a",
                        "TBC_germline.keep.gnmadF.cosmic.pathogenic.xlsx:md5,c2a934b08aa7fbfd81c6aba3b6ca8f8f",
                        "TBC_germline.keep.gnmadF.cosmic.pdf:md5,ef7b9a230caf546645bacf89d3f15a1b",
                        "TBC_germline.keep.gnmadF.cosmic.xlsx:md5,1d555beffd369a1b438b1fca4148e02b",
                        "TBC_germline.keep.gnmadF.maf.gz:md5,9289c1305179766867311c1dc4115a2a",
                        "TBC_germline.keep.gnmadF.noflags.maf.gz:md5,e8721975ebdb1508fcee79fabcf4558a",
                        "TBC_germline.keep.gnmadF.noflags.pdf:md5,97975dd9508d0105f24cdb19773335e8",
                        "TBC_germline.keep.gnmadF.noflags.xlsx:md5,ae03d7d73a73346c615987309941ff8a",
                        "TBC_germline.keep.gnmadF.pathogenic.maf.gz:md5,ac241cf6acb435a85233c63330b204cf",
                        "TBC_germline.keep.gnmadF.pathogenic.pdf:md5,c95bb6f36755a78890a56391520d37d3",
                        "TBC_germline.keep.gnmadF.pathogenic.xlsx:md5,4d52da387ad123621652c50d084a19bc",
                        "TBC_germline.keep.gnmadF.pdf:md5,9c047a6da3298144e8458e235899666c",
                        "TBC_germline.keep.gnmadF.xlsx:md5,41e1da6d177e518b8ee8b3b6a7ccd635",
                        "TBC_germline.keep.gnmadF.noflags.high.cgc.maf.gz:md5,cd97ad2bd772a953553538ec46aebb9a",
                        "TBC_germline.keep.gnmadF.noflags.high.cgc.pdf:md5,bca0e42a4eabe146a22fa7d25509d844",
                        "TBC_germline.keep.gnmadF.noflags.high.cgc.xlsx:md5,51b4a9a1ad1dde92a6801bf25659c69e",
                        "TBC_germline.keep.gnmadF.noflags.high.germ.maf.gz:md5,bea30dc469eb79b4875148bca34a66f7",
                        "TBC_germline.keep.gnmadF.noflags.high.germ.pdf:md5,82b367e43db36aa2d1664889a0c441a3",
                        "TBC_germline.keep.gnmadF.noflags.high.germ.xlsx:md5,fce09558b88f077c1db102448f686cf2",
                        "TBC_germline.keep.gnmadF.noflags.high.maf.gz:md5,eaa4a899a5ec0d2dffbabdb2f44b74b2",
                        "TBC_germline.keep.gnmadF.noflags.high.pdf:md5,90deb9b40f1a55b88ff9e60f0cbd0ea9",
                        "TBC_germline.keep.gnmadF.noflags.high.xlsx:md5,81c10ad6f76c2a2bf02d57b6ea8854e0",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.cgc.maf.gz:md5,4d8e4b9f88b002a2e5d3ae9432b4a588",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.cgc.pdf:md5,48b29edfcb3ef07254c870af79019eca",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.cgc.xlsx:md5,b7f25f4ef4d5e10c230a12f30d1402b0",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.germ.maf.gz:md5,e4c8d5d920f58b5f0d218daab6283927",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.germ.path.maf.gz:md5,9fa3864dc2700ab465b48d002b339957",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.germ.path.pdf:md5,f1b217b3e8d38e963da3772ae3f5cabf",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.germ.path.xlsx:md5,53ebebae8fbac416ff3091dfd045ed5f",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.germ.path_conf.maf.gz:md5,a1eff95091e21498692e3c130b83756b",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.germ.path_conf.xlsx:md5,f46142e72889f496308a1b044e0491e6",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.germ.pdf:md5,5cbba4595e0c426e17043000b357a130",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.germ.xlsx:md5,685ecc0421400cca40e7070139fb27df",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.maf.gz:md5,14de21bafd4b106fc47702ebcc327766",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.pdf:md5,83f2da571480595c67bd0e8c41ae207b",
                        "TBC_germline.keep.gnmadF.noflags.modhigh.xlsx:md5,f690edb1b5743ef03f32b72254d0a458"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-02-28T18:58:09.902797"
    }
}